plugins/inputs/mqtt_consumer/mqtt_consumer.go

```diff
@@ -115,23 +118,41 @@ func (m *MQTTConsumer) Start(acc telegraf.Accumulator) error {
 		return fmt.Errorf("MQTT Consumer, invalid QoS value: %d", m.QoS)
 	}
 
+	if m.ConnectionTimeout <= 0 {
+		return fmt.Errorf("MQTT Consumer, invalid connection_timeout value: %d", m.ConnectionTimeout)
+	}
+
 	opts, err := m.createOpts()
 	if err != nil {
 		return err
 	}
 
 	m.client = mqtt.NewClient(opts)
+
+	log.Printf("%v", m.connect())
+
+	return nil
+}
+
+func (m *MQTTConsumer) connect() error {
+
+	if m.in == nil {
+		m.in = make(chan mqtt.Message, 1000)
+	}
+
+	if m.done == nil {
+		m.done = make(chan struct{})
+	}
```

Comment:
Optional: I think if you put these channels in `Start` you won't have to guard them, we only call it once.